ui/hooks/src/use-login.new.ts

```diff
@@ -150,16 +195,30 @@ export function useLogin(onError?: (err: Error) => void) {
  * ```
  */
 export function useLogout() {
+  const [data, setData] = useState<unknown>(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<Error | null>(null);
   const sdk = getSDK();
-  return useMutation(
-    async () => {
-      const resp = await sdk.api.auth.signOut();
-      if (resp.data) {
-        return resp.data;
+
+  const logOut = useCallback(() => {
+    setIsLoading(true);
+    const logoutApiCall = async () => {
+      try {
+        const resp = await sdk.api.auth.signOut();
+        if (resp.data) {
+          setData(resp);
+          setIsLoading(false);
+        }
+      } catch (err) {
+        logError('use-logout', err);
+        setError(err);
       }
-    },
-    {
-      onError: (err: Error) => logError('use-logout', err),
-    },
-  );
+    };
+
+    logoutApiCall();
+
+    return { data, isLoading, error };
```

Comment:
you are trying to return the error in logout function via the state. Note that the state might not be updated yet. Maybe you want to just return the data? and return the error via the hook?